thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js

```diff
@@ -1,15 +1,229 @@
+      return Ember.isPresent(newNameField) || Ember.isPresent(newRrdField);
+    }
+  ),
+
+  /**
+   * Determines whether all fields are disabled (after submit)
+   * @method isFormDisabled
+   * @return {Boolean} isFormDisabled
+   */
+  isFormDisabled: Ember.computed(
+    'isExistingDashFieldDisabled',
+    'isCustomDashFieldDisabled',
+    function() {
+      return (this.get('isExistingDashFieldDisabled') && this.get('isCustomDashFieldDisabled'));
+    }
+  ),
+
+  /**
+   * Fetches an alert function record by name.
+   * Use case: when user names an alert, make sure no duplicate already exists.
+   * @method fetchAnomalyByName
+   * @param {String} functionName - name of alert or function
+   * @return {Promise}
+   */
+  fetchMetricsList(dataSet) {
+    const url = `/thirdeye-admin/metric-config/metrics?dataset=${dataSet}`
+    return fetch(url).then(res => res.json());
+  },
+
+  /**
+   * Triggers instant onboard for the generated metrics. If not triggered manually here,
+   * the onboard service runs every 15 minutes anyway.
+   * @method triggerInstantOnboard
+   * @return {Ember.RSVP.Promise}
+   */
+  triggerInstantOnboard() {
+    const url = '/autoOnboard/runAdhoc/AutometricsThirdeyeDataSource';
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url, { method: 'post' }).then(res => resolve(checkStatus(res)));
+    });
+  },
+
+  /**
+   * Generates/updates the metrics for an existing inGraphs dashboard
+   * https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Onboarding+ingraph+dashboards
+   * @method onboardNewDataset
+   * @param {Object} importObj - Request data containing dataset or metric name
+   * @return {Ember.RSVP.Promise}
+   */
+  onboardNewDataset(importObj) {
+    const postProps = {
+      method: 'post',
+      body: JSON.stringify(importObj),
+      headers: { 'content-type': 'Application/Json' }
+    };
+    const url = '/onboard/create';
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url, postProps).then(res => resolve(checkStatus(res)));
+    });
+  },
+
+  /**
+   * Generates/updates the metrics for an existing inGraphs dashboard
+   * https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Onboarding+ingraph+dashboards
+   * @method onboardMetricsToDataset
+   * @param {String} rrd - The metric RRD
+   * @return {Ember.RSVP.Promise}
+   */
+  onboardMetricsToDataset(rrd) {
+    const postProps = {
+      method: 'post',
+      body: JSON.stringify({ RRD: rrd }),
+      headers: { 'content-type': 'Application/Json' }
+    };
+    const url = '/onboard/create';
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url, postProps).then(res => resolve(checkStatus(res)));
+    });
+  },
+
+  /**
+   * Display results of new metrics onboarding
+   * @method prepareMetricListResult
+   * @param {Array} metricsList - List of new metrics
+   * @return {Ember.RSVP.Promise}
+   */
+  prepareMetricListResult(metricsList) {
+    if (!metricsList.Records.length) {
```

Comment:
nit: I think this can be cleaner with the logic in a setter. https://guides.emberjs.com/v2.14.0/object-model/computed-properties/#toc_setting-computed-properties
We can discuss this offline